docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,791 @@
+
+```csharp
+// MvvmLight
+this.RaisePropertyChanged(() => this.MyProperty);
+
+// Toolkit.Mvvm
+this.OnPropertyChanged(nameof(this.MyProperty));
+```
+
+#### Set<T> ( Expression, ref T, T )
+
+`Set(Expression, ref T, T)` does not have a like-for-like method signature replacement. 
+
+However, `SetProperty(ref T, T, string)` provides the same functionality with additional performance benefits. 
+
+```csharp
+// MvvmLight
+this.Set(() => this.MyProperty, ref this.myProperty, value);
+
+// Toolkit.Mvvm
+this.SetProperty(ref this.myProperty, value, nameof(this.MyProperty));
+```
+
+Note, string parameter is not required if the method is being called from the property's setter as it is inferred from the caller member name.
+
+#### Set<T> ( string, ref T, T )
+
+`Set<T>(string, ref T, T)` does not have a like-for-like method signature replacement. 
+
+However, `SetProperty<T>(ref T, T, string)` provides the same functionality with re-ordered parameters.
+
+```csharp
+// MvvmLight
+this.Set(nameof(this.MyProperty), ref this.myProperty, value);
+
+// Toolkit.Mvvm
+this.SetProperty(ref this.myProperty, value, nameof(this.MyProperty));
+```
+
+Note, string parameter is not required if the method is being called from the property's setter as it is inferred from the caller member name.
+
+#### Set<T> ( ref T, T, string )
+
+`Set<T>(ref T, T, string)` has a renamed direct replacement, `SetProperty<T>(ref T, T, string)`.
+
+```csharp
+// MvvmLight
+this.Set(ref this.myProperty, value, nameof(this.MyProperty));
+
+// Toolkit.Mvvm
+this.SetProperty(ref this.myProperty, value, nameof(this.MyProperty));
+```
+
+Note, string parameter is not required if the method is being called from the property's setter as it is inferred from the caller member name.
+
+### ObservableObject Properties
+
+#### PropertyChangedHandler
+
+`PropertyChangedHandler` does not have a direct replacement.
+
+To raise a property changed event via the `PropertyChanged` event handler, you need to call the `OnPropertyChanged` method instead. 
+
+```csharp
+// MvvmLight
+PropertyChangedEventHandler handler = this.PropertyChangedHandler;
+
+// Toolkit.Mvvm
+this.OnPropertyChanged(nameof(this.MyProperty));
+```
+
+## Migrating ViewModelBase
+
+The following steps focus on migrating your existing components which take advantage of the `ViewModelBase` of the MvvmLight Toolkit. 
+
+The Windows Community Toolkit MVVM framework provides an [`ObservableRecipient`](ObservableRecipient) type that provides similar functionality. 
+
+Below are a list of migrations that will need to be performed if being used in your current solution.
+
+### ViewModelBase Methods
+
+#### ICleanup.Cleanup ()
+
+There is no direct replacement for the `ICleanup` interface.
+
+However, the `ObservableRecipient` provides an `OnDeactivated` method which should be used to provide the same functionality as `Cleanup`. 
```

Comment:
We might want to mention that similarly to `ViewModelBase.Cleanup()`, the base implementation of `OnDeactivated()` also automatically unregisters the current recipient from all registered messages?

Reply:
Ah yeah, good point on this one. I've popped that info in there 👍